app_flutter/lib/agent_tile.dart

```diff
@@ -0,0 +1,138 @@
+  static const String _healthDetailsValue = 'details';
+  static const String _reserveTaskValue = 'reserve';
+
+  @override
+  Widget build(BuildContext context) {
+    final ThemeData theme = Theme.of(context);
+
+    final Agent agent = fullAgent.agent;
+    final AgentHealthDetails healthDetails = fullAgent.healthDetails;
+
+    return Card(
+      child: ListTile(
+        leading: CircleAvatar(
+          backgroundColor:
+              healthDetails.isHealthy ? Colors.green : theme.errorColor,
+          foregroundColor: Colors.white,
+          child: _getIconFromId(agent.agentId),
+        ),
+        title: Text(agent.agentId),
+        subtitle: AgentHealthDetailsBar(healthDetails),
+        trailing: PopupMenuButton<String>(
+          itemBuilder: (BuildContext context) => <PopupMenuEntry<String>>[
+            const PopupMenuItem<String>(
+              child: Text('Raw health details'),
+              value: _healthDetailsValue,
+            ),
+            const PopupMenuItem<String>(
+              child: Text('Authorize agent'),
+              value: _authorizeAgentValue,
+            ),
+            const PopupMenuItem<String>(
+              child: Text('Reserve task'),
+              value: _reserveTaskValue,
+            ),
+          ],
+          icon: Icon(Icons.more_vert),
+          onSelected: (String value) {
+            switch (value) {
+              case _healthDetailsValue:
+                _showHealthDetailsDialog(context, agent.healthDetails);
+                break;
+              case _authorizeAgentValue:
+                _authorizeAgent(context, agent);
+                break;
+              case _reserveTaskValue:
+                _reserveTask(context, agent);
+                break;
+              default:
+                throw Exception(
+                    '$value is not a valid operation on AgentTile popup menu');
+            }
+          },
+        ),
+      ),
+    );
+  }
+
+  Icon _getIconFromId(String agentId) {
+    if (agentId.contains('vm')) {
+      return Icon(Icons.dns);
+    } else if (agentId.contains('linux')) {
+      return Icon(Icons.android);
+    } else if (agentId.contains('mac')) {
+      return Icon(Icons.phone_iphone);
+    } else if (agentId.contains('windows')) {
+      return Icon(Icons.desktop_windows);
+    }
+
+    return Icon(Icons.device_unknown);
+  }
+
+  void _showHealthDetailsDialog(BuildContext context, String rawHealthDetails) {
+    // TODO(chillers): Add copy button when web has support. https://github.com/flutter/flutter/issues/46020
+    print(rawHealthDetails);
+
+    showDialog<SimpleDialog>(
+      context: context,
+      builder: (BuildContext context) => SimpleDialog(
+        children: <Widget>[
+          Text(rawHealthDetails),
+        ],
+      ),
+    );
+  }
+
+  Future<void> _authorizeAgent(BuildContext context, Agent agent) async {
+    final String token = await agentState.authorizeAgent(agent);
+    if (token != null) {
+      print(token);
```

Comment:
Is this print important?

Reply:
It allows for easily copying the token. Since web does not have clipboard support yet (https://github.com/flutter/flutter/issues/46020), and copying text does not work yet, this is a workaround. Eventually, this should create UI to show the token so it can be copied without opening devtools.

Added a TODO.